Can't import some modules

One example: [node-datachannel](https://github.com/murat-dogan/node-datachannel/).

Create a new folder, and run:

```javascript
$ npm install import-sync node-datachannel
$ node
> importSync = require('import-sync')
> importSync('node-datachannel')
Uncaught Error: 
        Failed to import from:
            node-datachannel.
        Options:
            {}
        Require error message:
            /tmp/tmp.cGqoObetIA/node_modules/import-sync/dist/cjs/import.js:1
Error: Cannot find module 'node-datachannel'
Require stack:
- /tmp/tmp.cGqoObetIA/node_modules/import-sync/dist/cjs/import.js
- /tmp/tmp.cGqoObetIA/node_modules/import-sync/dist/cjs/index.js
- <repl>
    at esmImport (/tmp/tmp.cGqoObetIA/node_modules/import-sync/dist/cjs/import.js:47:16)
    at importSync (/tmp/tmp.cGqoObetIA/node_modules/import-sync/dist/cjs/import.js:72:28)
    at REPL4:1
    at Script.runInThisContext (node:vm:122:12)
    at REPLServer.defaultEval (node:repl:594:29)
    
    at esmImport (/tmp/tmp.cGqoObetIA/node_modules/import-sync/dist/cjs/import.js:50:15)
    at importSync (/tmp/tmp.cGqoObetIA/node_modules/import-sync/dist/cjs/import.js:72:28)

> await import('node-datachannel')
[Module: null prototype] {
  Audio: [Function: Audio],
  DataChannel: [Function: DataChannel],
  ... // I.e. it imports with ESM just fine
}
```

Happens with Node 20.8.0 and 21.4.0 (latest). Any idea what's going on here? Node-fetch does indeed import as expected, but this module isn't even found for some reason.

--- a/package.json
+++ b/package.json
@@ -56,6 +56,6 @@
     "typescript": "^5.3.3"
   },
   "dependencies": {
-    "esm": "^3.2.25"
+    "@httptoolkit/esm": "^3.3.0"
   }
 }

--- a/src/httptoolkit-esm-types.d.ts
+++ b/src/httptoolkit-esm-types.d.ts
@@ -1,0 +1,4 @@
+declare module '@httptoolkit/esm' {
+    import * as Esm from 'esm';
+    export = Esm;
+}

--- a/src/import.ts
+++ b/src/import.ts
@@ -1,5 +1,6 @@
-import esm from 'esm';
+import esm from '@httptoolkit/esm';
 import fs from 'fs';
+
 import { ESMOptions, Options } from './types';
 import { findModuleFile, getCallerDirname } from './files';
 

--- a/src/import.ts
+++ b/src/import.ts
@@ -1,5 +1,4 @@
 import esm from '@httptoolkit/esm';
-import fs from 'fs';
 
 import { ESMOptions, Options } from './types';
 import { findModuleFile, getCallerDirname } from './files';
@@ -11,27 +10,7 @@ import { findModuleFile, getCallerDirname } from './files';
  */
 /* istanbul ignore next */
 const createEsmRequire = (esmOptions: ESMOptions) => {
-  /**
-   * Removes the error for "[ERR_REQUIRE_ESM]: require() of ES Module", as per
-   * - https://github.com/standard-things/esm/issues/855#issuecomment-657982788
-   */
-  require('module').Module._extensions['.js'] = (m: any, filename: string) => {
-    m._compile(fs.readFileSync(filename, 'utf-8'), filename);
-  };
-  const loader = esm(module, esmOptions);
-
-  const newModule = module.constructor.length > 1 ? module.constructor : loader('module');
-  const oldResolveFilename = newModule._resolveFilename;
-  /**
-   * Referencing
-   * - https://github.com/kenotron/esm-jest/commit/624b9524ee698f5cbd16ee2481dc4cd0dec52e42
-   * - https://github.com/standard-things/esm/issues/331#issuecomment-377056717
-   */
-  newModule._resolveFilename = function(request: string, parent: any, isMain: boolean) {
-    const newRequest = request.startsWith('node:') ? request.substring(5) : request;
-    return oldResolveFilename.call(this, newRequest, parent, isMain);
-  };
-  return loader;
+  return esm(module, esmOptions);
 };
 
 /**

--- a/src/types.ts
+++ b/src/types.ts
@@ -1,4 +1,4 @@
-import esm from 'esm';
+import esm from '@httptoolkit/esm';
 
 export type ESMOptions = Parameters<typeof esm>[1];
 

